### DISCUSSION TITLE
Using with HTTPS

### DISCUSSION THREAD

**Discussion by LimitedGlove** [TannerGabriel/WebRTC-Video-Broadcast#71] [Q&A] [answered]
Is there a way to enable https? I am trying to set this up on a VPS so I can have a livestream on my website. Everything works great locally, but when I try to access the pages from the VPS on port 4000 I get an SSL Protocol Error. I can access the pages on port 443, but the JavaScript doesn't work.

**Comment by TannerGabriel** [comment, marked as answer]
The repo does not include a configuration for HTTPS, so you would need to add it yourself. You will need a certificate and a secure WebSockets connection. Something like:

```javascript
const express = require("express");
const app = express();
const fs = require('fs');
const https = require('https');

// SSL certificate
const options = {
  key: fs.readFileSync('path/to/privatekey.key'),
  cert: fs.readFileSync('path/to/certificate.crt')
};

let broadcaster;
const port = 4000;

const server = https.createServer(options, app);
```

And then change from `ws` to `wss` for the `broadcast` and `watcher`. If establishing the peer connection fails, consider adding a Turn server.

Another option for the certificate would be using the [Nginx LetsEncrypt companion](https://github.com/nginx-proxy/acme-companion) in Docker.
